From 26594cd257e38014b4912ab1e1f045a83258c9e9 Mon Sep 17 00:00:00 2001
From: Fabio Estevam <fabio.estevam@nxp.com>
Date: Fri, 15 Dec 2017 00:24:58 -0200
Subject: [PATCH 1/2] ARM: dts: imx6sx: Add support for PCI power domain

commit 1f1b7ec6714e08c9ab4243146ff8b572a25d1837 upstream

Previously PCI support was working because the bootloader has previously
powered up the PCI power domain.

Represent the PCI power domain, so that PCI is functional without
relying on the PCI support from the bootloader.

Tested on a imx6sx-sdb board with no PCI support in the bootloader.

Reported-by: Abel Vesa <abel.vesa@nxp.com>
Signed-off-by: Fabio Estevam <fabio.estevam@nxp.com>
Reviewed-by: Lucas Stach <l.stach@pengutronix.de>
Signed-off-by: Shawn Guo <shawnguo@kernel.org>
[Quanyang: some modifications at context]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/arm/boot/dts/imx6sx.dtsi |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index ec94c22..49c9334 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -880,6 +880,16 @@
 						"lcdif_axi", "lcdif2_pix", "csi_mclk";
 				pcie-phy-supply = <&reg_pcie_phy>;
 				#power-domain-cells = <1>;
+				pgc {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					pd_pci: power-domain@3 {
+						reg = <3>;
+						#power-domain-cells = <0>;
+						power-supply = <&reg_pcie>;
+					};
+				};
 			};
 
 			iomuxc: iomuxc@020e0000 {
@@ -1552,7 +1562,7 @@
 				 <&clks IMX6SX_CLK_DISPLAY_AXI>;
 			clock-names = "pcie", "pcie_bus", "pcie_phy", "pcie_inbound_axi";
 			pcie-phy-supply = <&reg_pcie_phy>;
-			power-domains = <&gpc 2>;
+			power-domains = <&pd_pci>;
 			fsl,max-link-speed = <2>;
 			status = "disabled";
 		};
-- 
1.7.9.5

